core/translate/expr.rs

```diff
@@ -698,7 +698,93 @@ pub fn translate_expr(
             }
             Ok(target_register)
         }
-        ast::Expr::Case { .. } => todo!(),
+        ast::Expr::Case {
+            base,
+            when_then_pairs,
+            else_expr,
+        } => {
+            // There's two forms of CASE, one which does quality against the when values:
+            //   CASE [expr] WHEN [value] ELSE [value] END
+            // And one which evaluates a series of boolean predicates:
+            //   CASE (WHEN [bool_expr] THEN [value])+ END
+            // This just changes which sort of branching instruction to issue, after we
+            // generate the expression if needed.
+            let return_label = program.allocate_label();
+            let mut next_case_label = program.allocate_label();
+            let base_reg = base.as_ref().map(|_| program.alloc_register());
+            let expr_reg = program.alloc_register();
+            if let Some(base_expr) = base {
+                translate_expr(
+                    program,
+                    referenced_tables,
+                    base_expr,
+                    base_reg.unwrap(),
+                    precomputed_exprs_to_registers,
+                )?;
+            };
+            for (when_expr, then_expr) in when_then_pairs {
+                translate_expr(
+                    program,
+                    referenced_tables,
+                    when_expr,
+                    expr_reg,
+                    precomputed_exprs_to_registers,
+                )?;
+                match base_reg {
+                    Some(base_reg) => program.emit_insn_with_label_dependency(
```

Comment:
Even though you have done so above, I would still comment both of these match arms with what they are doing. Saves someone 30 seconds when they look at this the next time :)